File: Components/ASP.NET/VB/ASPxSchedulerDemos/DevExpress/ASPxSchedulerForms/SchedulerMessageBox.ascx.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Web.UI
Imports DevExpress.XtraScheduler
Imports DevExpress.Web
Imports DevExpress.Web.ASPxScheduler
Imports DevExpress.Web.ASPxScheduler.Internal
Imports System.Collections
Imports System.Collections.Generic
Imports DevExpress.XtraScheduler.Localization
Imports DevExpress.Web.ASPxScheduler.Localization
Imports DevExpress.Utils
Imports System.Web.UI.WebControls

Partial Public Class SchedulerMessageBox
	Inherits SchedulerMessageBoxBase
	Public Overrides ReadOnly Property ClassName() As String
		Get
			Return "ASPxSchedulerMessageBox"
		End Get
	End Property

	Protected Overrides Sub OnLoad(ByVal e As EventArgs)
		MyBase.OnLoad(e)
		Localize()
	End Sub
	Private Sub Localize()
		btnOk.Text = ASPxSchedulerLocalizer.GetString(ASPxSchedulerStringId.Form_ButtonOk)
		btnCancel.Text = ASPxSchedulerLocalizer.GetString(ASPxSchedulerStringId.Form_ButtonCancel)
		btnOk.Wrap = DefaultBoolean.False
		btnCancel.Wrap = DefaultBoolean.False
	End Sub
	Protected Overrides Function GetChildEditors() As ASPxEditBase()
		Return New ASPxEditBase() { }
	End Function
	Protected Overrides Function GetChildButtons() As ASPxButton()
		Dim buttons() As ASPxButton = { btnOk, btnCancel }
		Return buttons
	End Function
	Protected Overrides Function GetChildControls() As Control()
		Return New Control() { root, lblMessage }
	End Function
	Protected Overrides Function GetDefaultButton() As WebControl
		Return btnOk
	End Function
End Class

File: Components/ASP.NET/CS/ASPxSchedulerDemos/DevExpress/ASPxSchedulerForms/SchedulerMessageBox.ascx.cs
using System;
using System.Web.UI;
using DevExpress.XtraScheduler;
using DevExpress.Web;
using DevExpress.Web.ASPxScheduler;
using DevExpress.Web.ASPxScheduler.Internal;
using System.Collections;
using System.Collections.Generic;
using DevExpress.XtraScheduler.Localization;
using DevExpress.Web.ASPxScheduler.Localization;
using DevExpress.Utils;
using System.Web.UI.WebControls;

public partial class SchedulerMessageBox : SchedulerMessageBoxBase {
    public override string ClassName { get { return "ASPxSchedulerMessageBox"; } }
      
    protected override void OnLoad(EventArgs e) {
        base.OnLoad(e);
        Localize();
    }
    void Localize() {
        btnOk.Text = ASPxSchedulerLocalizer.GetString(ASPxSchedulerStringId.Form_ButtonOk);
        btnCancel.Text = ASPxSchedulerLocalizer.GetString(ASPxSchedulerStringId.Form_ButtonCancel);
        btnOk.Wrap = DefaultBoolean.False;
        btnCancel.Wrap = DefaultBoolean.False;
    }
    protected override ASPxEditBase[] GetChildEditors() {
        return new ASPxEditBase[] { };
    }
    protected override ASPxButton[] GetChildButtons() {
        ASPxButton[] buttons = new ASPxButton[] {
            btnOk, btnCancel
        };
        return buttons;
    }
    protected override Control[] GetChildControls() {
        return new Control[] { root, lblMessage };
    }
    protected override WebControl GetDefaultButton() {
        return btnOk;
    }
}
